### DISCUSSION TITLE
How to annotate images with small objects?

### DISCUSSION THREAD

**Discussion by bit-scientist** [obss/sahi#656] [Q&A] [answered]
First of all, thank you a lot for this practical repo, it is quite wonderful.

I have a dataset of images collected using a camera and the task is to detect defects on industrial fabric surface. As it turns out, the size of almost all defects are very small compared to the image size. (around 40x40, even 20x20). However, I don't know how I should do the annotation. I reviewed the tutorials and some issues to see if there were related questions, but unfortunately there isn't any. So here goes my question:

1. While annotating, do I need to label every single defect that is very tiny or can I just label larger defects and hope that model will manage to detect tiny defects once SAHI is used?
The reason why I am asking is that I saw an image of many small cars detected with SAHI, but not detected without it, and I can't believe that label files used for training contained the annotations for all those small objects . 
2.  Camera is supposed to capture 1 image per second with 1920 x 1080 resolution, and the system should detect small defects on that image in less than 1s. How do I prepare input image for SAHI detection in this scenario? 

I will ask the rest of my questions once I get feedback. 😄 

Thank you!

**Comment by fcakyon** [comment, marked as answer]
Hello @bit-scientist, thanks for your interest in `SAHI`!

You use `SAHI` in 2 settings:
1. Dont annotate smaller objects in the dataset, perform standard fine-tuning (without any crop augmentation), then perform slicing aided inference
2. Annotate each small object in the dataset, perform slicing aided fine-tuning (with some crop augmentation), then perform slicing aided inference

1st and 2nd settings give up to `6.8 AP` and `14.5 AP` boost over standard inference on the Visdrone19-Detection test-dev set, respectively. Please refer to [official sahi paper](https://arxiv.org/pdf/2202.06934.pdf) for more detail.
